packages/core/src/tracing/sampling.ts

```diff
@@ -27,7 +27,21 @@ export function sampleSpan(
   // work; prefer the hook if so
   let sampleRate;
   if (typeof options.tracesSampler === 'function') {
-    sampleRate = options.tracesSampler(samplingContext);
+    sampleRate = options.tracesSampler({
+      ...samplingContext,
+      inheritOrSampleWith(fallbackSampleRate) {
```

Comment:
```suggestion
      inheritOrSampleWith: (fallbackSampleRate) => {
```

is this maybe slightly cleaner regarding `this` context? 🤔